monash-human-power/dashboard
# Add systemd files for dashboard client and server

## Description

Add service unit files to allow dashboard client and server to be run by systemd daemon. This would help automatically start the client and server when raspberry pi boots up and will auto-restart either process if it crashes.

I made seperate unit files for client and server, because they are independent of each other when a crash occurs. Keeping them in seperate unit files means if one crashes we do not unnecessarily restart both.

## Screenshots

N/A

## Steps to Test

1) Grab a raspberry pi
2) Add the unit files by running the `install.sh` script in `client/service` and `server/service`. 
Hint: You may need to use `chmod +x install.sh` to set execution permissions

3) Check dashboard_client.service and dashboard_server.service is moved in the systemd user unit files directory. Usually this would be `~/.config/systemd/user/`
4) Follow the Readme in the `client/service` and `server/service` directories to enable and start each service.
5) Once started, check the logs for any errors.
6) If no error, you should be able to interact with dashboard from your local computer if it is connected to the same network as the pi. The url would be `{IP address of pi}:3000`.
